mmv1/third_party/terraform/services/bigquery/iam_bigquery_dataset.go

```diff
@@ -222,19 +222,19 @@ func iamMemberToAccess(member string) (string, string, error) {
 			return "groupByEmail", pieces[1], nil
 		case "domain":
 			return "domain", pieces[1], nil
+		case "iamMember":
+			return "iamMember", pieces[1], nil
```

Comment:
If I may add a small remark:
I think this switch/case mapping was heavily inspired by the V1 prefixes used by IAM:
https://cloud.google.com/iam/docs/principal-identifiers
(I only found these "deleted" information in this doc for instance).
If you're not planning to move to V2 API anytime soon, might make sense to follow the same logic and use principal/principalSet.
Moreover pieces[1] won't work for my use case here since I need the API to be fed with the entire string (not the cut string), so the line 237 was actually correct.
I mean that the API is expecting "iamMember":"principal://iam.googleapis.com/locations/global/workforcePools/POOL_ID/subject/SUBJECT_ATTRIBUTE_VALUE" 
but NOT
"iamMember":"//iam.googleapis.com/locations/global/workforcePools/POOL_ID/subject/SUBJECT_ATTRIBUTE_VALUE"

Reply:
@tgmof thanks for the feedback.
I'm not understanding. Can you please show me a use case where this wouldn't work so I can better understand?
Thanks

Reply:
Sure!
I am trying to give rights to a list of Federated Identity users that all have this format:
principal://iam.googleapis.com/locations/global/workforcePools/POOL_ID/subject/SUBJECT_ATTRIBUTE_VALUE

It works via GCP Console UI (where I need to put the full string above) but not via Terraform. It also works via this API:
https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets/patch
<img width="229" alt="image" src="https://github.com/GoogleCloudPlatform/magic-modules/assets/31918716/a4d85f6c-0e8f-4524-b41d-b4011a296564">

I understand that, for instance, the API expects "groupByEmail": "some-gcp-group-email" without the leading "group:" but for Federated Identity iamMember it's different, the API needs to full string:
"iamMember": "principal://iam.googleap..." and the principal: should not be cut off into pieces[0] (ignored) and pieces[1] (passed to the API) like it is in your PR proposal:
https://cloud.google.com/iam/docs/principal-identifiers
https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets/patch

Reply:
pieces[1] will return the full principal (principal://.....)
Please, notice that we splitN 2. So we're splitting the string at most in two parts. Meaning that in case of "iamMember:principal://...." it will return `pieces[0]` = `iamMember` and `pieces[1]` = `principal://iam....`

Reply:
Got it now, thank you for having taken the time to answer! I wrongly assumed that it needed to be passed without iamMember: to Terraform (because the terraform argument itself it called member, but anyway wrong assumption).

Anyway, I guess your commit will also re-generate the doc here https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_dataset_iam#google_bigquery_dataset_iam_member so I won't have to make any assumption after you merge your PR.
Thanks again!

Reply:
Thanks for double checking @tgmof !
Yes, the PR will also generate the new doc.